### PR TITLE
EXAMPLEAND-30 implement implicit authentication example

### PR DESCRIPTION
**Pull request by marcin-onegini**
…plicit resource call

### CODE REVIEW

File: app/src/main/java/com/onegini/mobile/exampleapp/model/ImplicitUserDetails.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.onegini.mobile.exampleapp.model;
+
+import com.google.gson.annotations.SerializedName;
+
+@SuppressWarnings("unused")
```
Comment (robertszuba) [inline app/src/main/java/com/onegini/mobile/exampleapp/model/ImplicitUserDetails.java]: The warning should be suppressed on the property level (the minimum required)

File: app/src/main/java/com/onegini/mobile/exampleapp/model/ApplicationDetails.java [hunk 2]
```diff
@@ -29,19 +28,19 @@
   @SerializedName("application_version")
   private String applicationVersion;
 
-  public String getApplicationIdentifier() {
+  private String getApplicationIdentifier() {
```
[the remark this answers is not in the document]
Comment (marcin-onegini) [reply app/src/main/java/com/onegini/mobile/exampleapp/model/ApplicationDetails.java]: done

File: app/src/main/java/com/onegini/mobile/exampleapp/network/ImplicitUserService.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  private ImplicitUserService(final Context context) {
+    applicationDetailsRetrofitClient = SecureResourceClient.prepareSecuredImplicitUserRetrofitClient(ImplicitUserClient.class, context);
+    applicationDetailsRetrofit2Client = SecureResourceClient.prepareSecuredImplicitUserRetrofit2Client(ImplicitUserClient2.class, context);
+  }
+
+  public Observable<ImplicitUserDetails> getImplicitUserDetails(final boolean useRetrofit2) {
+    return getObservable(useRetrofit2)
+        .subscribeOn(Schedulers.io())
+        .observeOn(AndroidSchedulers.mainThread())
+        .unsubscribeOn(Schedulers.io());
```
[the remark this answers is not in the document]
Comment (marcin-onegini) [reply app/src/main/java/com/onegini/mobile/exampleapp/network/ImplicitUserService.java]: done

File: app/src/main/java/com/onegini/mobile/exampleapp/network/client/SecureResourceClient.java [hunk 4]
```diff
@@ -31,51 +32,76 @@
...
+  }
 
-    return retrofit.create(clazz);
+  /*
+   * implicit user clients
+   */
+
+  // preparing the client using Retrofit 1.9
+  public static <T> T prepareSecuredImplicitUserRetrofitClient(final Class<T> clazz, final Context context) {
+    return prepareSecuredRetrofitClient(clazz, context, OneginiSDK.getOneginiClient(context).getUserClient().getImplicitResourceOkHttpClient());
```
[the remark this answers is not in the document]
Comment (marcin-onegini) [reply app/src/main/java/com/onegini/mobile/exampleapp/network/client/SecureResourceClient.java]: done

File: app/src/main/java/com/onegini/mobile/exampleapp/model/ImplicitUserDetails.java (hunk 1 above)
Comment (marcin-onegini) [reply app/src/main/java/com/onegini/mobile/exampleapp/model/ImplicitUserDetails.java]: done

**Pull request merged**